molSimplify/Classes/mol3D.py

```diff
@@ -5683,6 +5685,183 @@ def get_geometry_type(self, dict_check=False, angle_ref=False,
+                #if the RMSDs are close, flag this in the summary and classify on second criterion
+                if np.abs(summary[geotype][0] - current_rmsd) < close_dev:
+                    close_rmsds = True
+                    if summary[geotype][1] < summary[geometry][1]:
+                        #classify based on largest singular deviation
+                        current_rmsd = summary[geotype][0]
+                        geometry = geotype
+                else:
+                    current_rmsd = summary[geotype][0]
+                    geometry = geotype
+
+        results = {
+            "geometry": geometry,
+            "rmsd": current_rmsd,
+            "summary": summary,
+            "hapticity": hapt,
+            "close_rmsds": close_rmsds
+        }
+        return results
+
+    def dev_from_ideal_geometry(self, ideal_polyhedron):
+        """
+        Return the minimum RMSD between a geometry and an ideal polyhedron (with the same average bond distances).
+        Enumerates all possible indexing of the geometry. As such, only recommended for small systems.
+
+        Parameters
+        ----------
+            ideal_polyhedron: np.array of 3-tuples of coordinates
+                Reference list of points for an ideal geometry
+
+        Returns
+        -------
+            rmsd: float
+                Minimum root mean square distance between the fed geometry and the ideal polyhedron
+            single_dev: float
+                Maximum distance between any paired points in the fed geometry and the ideal polyhedron.
+        """
+
+        metal_idx = self.findMetal()
+        if len(metal_idx) == 0:
+            raise ValueError('No metal centers exist in this complex.')
+        elif len(metal_idx) != 1:
+            raise ValueError('Multimetal complexes are not yet handled.')
+        temp_mol = self.get_first_shell()[0]
+        fcs_indices = temp_mol.get_fcs(max6=False)
+        # remove metal index from first coordination shell
+        fcs_indices.remove(temp_mol.findMetal()[0])
+
+        if len(fcs_indices) != len(ideal_polyhedron):
+            raise ValueError('The coordination number differs between the two provided structures.')
+
+        #have to redo getting metal_idx with the new mol after running get_first_shell
+        #want to work with temp_mol since it has the edge and sandwich logic implemented to replace those with centroids
+        metal_atom = temp_mol.getAtoms()[temp_mol.findMetal()[0]]
+        fcs_atoms = [temp_mol.getAtoms()[i] for i in fcs_indices]
+        #construct a np array of the non-metal atoms in the FCS
+        distances = []
+        positions = np.zeros([len(fcs_indices), 3])
+        for idx, atom in enumerate(fcs_atoms):
+            distance = atom.distance(metal_atom)
+            distances.append(distance)
+            positions[idx, :] = np.array(atom.coords()) - np.array(metal_atom.coords()) #shift so the metal is at (0, 0, 0)
+
+        def permutations(list):
```

Comment:
Which I have done in my recent changes

Reply:
I believe that `get_first_shell` is limited to a maximum coordination number of 6: it uses the function `obtain_truncation_metal` to get the first shell (before applying any edge/sandwich logic), which determines neighbors using the `getBondedAtomsSmart` function. This function, as shown in change [https://github.com/hjkgrp/molSimplify/pull/213/commits/79272c199d4c67ccd4bcb43c2e11b57195d113c8](url), assumes a maximum coordination number of 6. If the default behavior of `getBondedAtomsSmart` is set to not assume a maximum of 6-coordinated structures, some of the tests will fail for geometry checks, so I do not want to change that behavior. However, changing the `oct` argument to `False` will allow for `get_geometry_type_distance` to classify geometries with coordination numbers higher than 6, which I have tested. As seen in the commit I referenced earlier, I can introduce a workaround variable to allow for different behavior, but I do not want to add that argument to more functions than needed (since readability on those extra arguments would be difficult to interpret). I would argue that we should change the test cases to add `oct=True` to `getBondedAtomsSmart`, but I am not sure if that would also require adding arguments to many functions to ensure that the test cases work as intended.

Reply:
I just spent some time trying to understand why these 4 test cases:
- `test_geocheck_oct[broken_ligands] `
- `test_geocheck_oct[H_transfer] `
- `test_geocheck_oct[rotational_group]` 
- `test_geocheck_oct[iodine_sulfur]`  

fail when we set `oct=False` as default in `getBondedAtomsSmart` and got nowhere... My recommendation for now is just to merge what you have and address this issue during the hackathon on higher coordination complexes.